src/string.cr

```diff
@@ -522,18 +522,19 @@ class String
     end
 
     negative = false
-    found_digit = false
-    mul_overflow = ~0_u64 / base
 
     # Check + and -
     case ptr.value.unsafe_chr
-    when '+'
-      ptr += 1
```

Comment:
I'm pretty sure numbers like `+123` are less common than a negative number so I've moved this below the `'-'` check.